client/my-sites/comments/comment/comment-edit.jsx

```diff
@@ -38,15 +43,33 @@ export class CommentEdit extends Component {
 	};
 
 	state = {
-		authorDisplayName: '',
-		authorUrl: '',
-		commentContent: '',
+		authorDisplayName: this.props.authorDisplayName || '',
+		authorUrl: this.props.authorUrl || '',
+		commentContent: this.props.commentContent || '',
+		commentDate: this.props.commentDate || '',
+		isDatePopoverVisible: false,
+		tmpCommentDate: '',
 	};
 
-	componentWillMount() {
-		const { authorDisplayName, authorUrl, commentContent } = this.props;
-		this.setState( { authorDisplayName, authorUrl, commentContent } );
-	}
+	datePopoverButtonRef = createRef();
+
+	toggleDatePopover = () =>
+		this.setState( ( { commentDate, isDatePopoverVisible } ) => ( {
+			isDatePopoverVisible: ! isDatePopoverVisible,
+			tmpCommentDate: isDatePopoverVisible ? '' : commentDate,
+		} ) );
+
+	cancelCommentDataValueChange = () =>
+		this.setState( ( { tmpCommentDate } ) => ( {
+			commentDate: tmpCommentDate,
+			isDatePopoverVisible: false,
+			tmpCommentDate: '',
```

Comment:
`tmp` here is an awkward term because it denotes the _longevity_ of the data but not the intent. could we think of a more descriptive name like `storedCommentDate` or something that indicates its _use_?